### ISSUE TITLE
Fix FP S2189 ('no-infinite-loop'): Ignore condition variables called on or passed over

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#2866]
We are currently raising false positives on variables of loop conditions that could possibly be subject to mutation through function and methods calls:

```javascript
let foo = { /* ... */ };
while (foo) { // FP
  foo.doSomething(); // `doSomething` could modify `foo`
}
```

We should assume that such variables are always modified in order to kill the noise.

Reported in https://community.sonarsource.com/t/l3989-false-positive-in-loopsshouldnotbeinfinitecheck-should-support-guard-clause
